src/main/scala/pipedsl/codegen/bsv/BluespecGeneration.scala

```diff
@@ -631,6 +632,9 @@ object BluespecGeneration {
             BUOp("!", BFromMaybe(BBoolLit(true),
               bsInts.getSpecCheck(specTable, getSpecIdVal, stgSpecOrder))))
         //also need these in case we're waiting on responses we need to dequeue
+        case IStageClear() =>
```

Comment:
It doesn't seem like you modified the KillLogic to generate an `OR` -> i.e., the KillConds list is implicitly an `AND` of conditions, but we want the stage to be killed if `IStageClear()` is true, OR the rest of the conditions are all true.